packages/scene-composer/src/components/panels/SceneHierarchyPanel/components/SceneHierarchyTree/SceneHierarchyTreeItem.tsx

```diff
@@ -49,6 +58,16 @@ const SceneHierarchyTreeItem: FC<SceneHierarchyTreeItemProps> = ({
   const { searchTerms } = useSceneHierarchyData();
   const isSearching = searchTerms !== '';
 
+  useEffect(() => {
+    /**
+     * Use default state only if node is not expanded
+     * If node is already expanded, then skip this
+     *  */
+    if (!expanded) {
+      setExpanded(defaultExpanded);
+    }
+  }, [defaultExpanded, pathFromSelectedToRoot]);
```

Comment:
do we need `pathFromSelectedToRoot` as a dependency here?

Reply:
Yes, it is needed to update the default state when the selected node changes (which in turn updates pathFromSelectedToRoot).

Reply:
but if `pathFromSelectedToRoot` changes that impacts the current item, then `defaultExpanded` (the expended value from props) will be changed correct?

Reply:
There is a case where if user manually collapses a node which was expanded, the defaultExpanded state does not changes. In this case, it relies on pathFromSelectedToRoot.

Reply:
in that case, will `pathFromSelectedToRoot` be changed? isn't `pathFromSelectedToRoot` only depending on current selection?

Reply:
I missed to mention that selection change is the next step after user manually collapses the node. And in that case, `pathFromSelectedToRoot` is updated.